streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -407,9 +413,13 @@ public void testNoMessagesSentExceptionFromOverlappingPatterns() throws Exceptio
         IntegrationTestUtils.produceValuesSynchronously(FOO_TOPIC, Arrays.asList(fooMessage), producerConfig, mockTime);
 
         final Properties consumerConfig = TestUtils.consumerConfig(CLUSTER.bootstrapServers(), StringDeserializer.class, StringDeserializer.class);
+        try {
+            IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, DEFAULT_OUTPUT_TOPIC, 2, 5000);
```

Comment:
add `fail()` in next line to make sure we hit the timeout and throw an `AssertionError`

Reply:
`fail()` will also throw an `AssertionError`, right?

Reply:
Good point -- maybe throw an `new RuntimeException` ? This should not be caught in the `catch()` below.

Reply:
Ack, addressed this comment in a new commit.